Source/Mosa.DeviceSystem/Bitmap.cs

```diff
@@ -20,24 +20,21 @@ public static Image CreateImage(byte[] data)
 			var height = stream.Read32(0x16);
 			var dataSectionOffset = stream.Read32(0xA);
 
-			var image = new Image(width, height);
+			var image = new Image(width, height, bpp);
 
-			int x = 0;
-			int y = height - 1;
-
-			for (var i = dataSectionOffset; i < dataSectionOffset + (width * height * bpp); i += bpp)
+			// TODO: Support for other color depths
+			for (uint y = 0; y < height - 1; y++)
 			{
-				var color = (bpp == 4) ? stream.Read32(i) : (int)(stream.Read24(i) | 0xFF000000); // 32 & 24 bit
-
-				image.SetColor(x, y, color);
-
-				x++;
-
-				if (x == width)
+				for (uint x = 0; x < width - 1; x++)
 				{
-					x = 0;
-					y--;
+					dataSectionOffset += x * bpp;
```

Comment:
As you're incrementing dataSectionOffset during each iteration of the loop, you don't need to multiply by x, you can simply += bpp